lib/internal/encoding.js

```diff
@@ -396,13 +397,14 @@ function makeTextDecoderICU() {
         flags |= options.ignoreBOM ? CONVERTER_FLAGS_IGNORE_BOM : 0;
       }
 
-      // Only support fast path for UTF-8 without FATAL flag
-      const fastPathAvailable = enc === 'utf-8' && !(options?.fatal);
+      // Only support fast path for UTF-8.
+      const fastPathAvailable = enc === 'utf-8';
```

Comment:
Do we still need this or can we just do in line 407?

this[kUTF8FastPath] = enc === 'utf-8';

Reply:
Yup, we can inline it.